source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
Well, OpBitCast doesn't say it's limited to Kernel or Addresses and it clearly states that it can bitcast pointers, so it should be possible even for Logical/shaders.

Also, where exactly in the spec does it say that you have to use Logical for shaders? I can't seem to find this.

Either way, there are more instructions that can result in pointers even for Logical/shaders (2.16.1 is incomplete as I said).

So, it would be better to just remove spvOpcodeIsPointer/spvOpcodeIsPotentialPointer and just validate that the input pointer operand is actually a pointer.